src/java/org/apache/cassandra/io/sstable/format/SortedTableWriter.java

```diff
@@ -153,8 +153,14 @@ public static CompressionParams compressionFor(final OperationType opType, Table
                 case fast:
                     if (!compressor.recommendedUses().contains(ICompressor.Uses.FAST_COMPRESSION))
                     {
-                        // The default compressor is generally fast (LZ4 with 16KiB block size)
-                        compressionParams = CompressionParams.DEFAULT;
+                        compressionParams = CompressionParams.FAST;
+                        break;
+                    }
+                    // else fall through
+                case adaptive:
+                    if (!compressor.recommendedUses().contains(ICompressor.Uses.FAST_COMPRESSION))
```

Comment:
That's not going to be a problem because the adaptive compressor configured for compaction does *not* advertise itself as FAST compressor, so it cannot be picked here.

Reply:
The `table` mode does preserve the compressor, and will end up with the wrong set-up.

Reply:
Reworked according to your comment below (added new Compressor method)